### PR TITLE
[FIX] Fixes floating blood on non-existent hands: My damn feet and hands are bloody!

### PR DESCRIPTION
**Pull request by Spaghetti-bit**
## What Does This PR Do

Adds checks to overlays to make sure we don't get floating blood.

Also adds four new overlay layers for handling specific missing limbs for blood overlays.
Both hands and both feet.

## Why It's Good For The Game

Unintended oversight. Fixes said oversight.

## Images of changes

image
image

## Testing

Loaded it locally, smothered a skrell in blood. Nugget.

## Changelog
:cl:
fix: Floating blood overlays.
/:cl:

### CODE REVIEW

File: code/modules/mob/living/carbon/human/human_update_icons.dm [hunk 1]
```diff
@@ -259,7 +259,7 @@ GLOBAL_LIST_EMPTY(damage_icon_parts)
 			var/u2_icon = U2.sprite_sheets && (dna.species.sprite_sheet_name in U2.sprite_sheets) ? U2.sprite_sheets[dna.species.sprite_sheet_name] : U2.icon
 			underwear_standing.Blend(new /icon(u2_icon, "us_[U2.icon_state]_s"), ICON_OVERLAY)
 
-	if(socks && dna.species.clothing_flags & HAS_SOCKS)
+	if(socks && dna.species.clothing_flags & HAS_SOCKS && (get_organ("l_leg") && get_organ("r_leg"))) // Check if the human has both legs before going on adding socks.
```
Comment (lewcc) [inline code/modules/mob/living/carbon/human/human_update_icons.dm:262]: this seems to be from the other PR?

Comment (Spaghetti-bit) [reply code/modules/mob/living/carbon/human/human_update_icons.dm:262]: It is.... not intended, give me a second.

File: code/__DEFINES/misc_defines.dm [hunk 2]
```diff
@@ -182,22 +182,26 @@
...
+#define MARKINGS_LAYER			41
+#define INTORGAN_LAYER			40
+#define UNDERWEAR_LAYER			39
+#define MUTATIONS_LAYER			38
+#define H_DAMAGE_LAYER			37
+#define UNIFORM_LAYER			36
+#define ID_LAYER				35
+#define HANDS_LAYER				34	//Exists to overlay hands over jumpsuits
+#define SHOES_LAYER				33
+#define L_FOOT_BLOOD_LAYER		33	// Blood overlay separation Right-Foot
```
Comment (lewcc) [inline code/__DEFINES/misc_defines.dm]:
```suggestion
#define L_FOOT_BLOOD_LAYER		33	// Blood overlay separation Left-Foot
```

File: code/modules/mob/living/carbon/human/human_update_icons.dm [hunk 3]
```diff
@@ -677,17 +679,24 @@ GLOBAL_LIST_EMPTY(damage_icon_parts)
...
-			overlays_standing[GLOVES_LAYER]	= bloodsies
+		if(blood_DNA) // Checks for hands to make sure we don't get mysterious floating blood.
+			if(get_organ("l_hand"))
+				var/mutable_appearance/bloodsies_left 	= mutable_appearance(dna.species.blood_mask, "bloodyhand_left", layer = -L_HAND_BLOOD_LAYER)
+				bloodsies_left.color = feet_blood_color
+				overlays_standing[L_HAND_BLOOD_LAYER] 	= bloodsies_left
+			if(get_organ("r_hand"))
+				var/mutable_appearance/bloodsies_right 	= mutable_appearance(dna.species.blood_mask, "bloodyhand_right", layer = -R_HAND_BLOOD_LAYER)
+				bloodsies_right.color = feet_blood_color
+				overlays_standing[R_HAND_BLOOD_LAYER] 	= bloodsies_right
```
Comment (lewcc) [inline code/modules/mob/living/carbon/human/human_update_icons.dm:696]:
```suggestion
				var/mutable_appearance/bloodsies_left = mutable_appearance(dna.species.blood_mask, "bloodyhand_left", layer = -L_HAND_BLOOD_LAYER)
				bloodsies_left.color = feet_blood_color
				overlays_standing[L_HAND_BLOOD_LAYER] = bloodsies_left
			if(get_organ("r_hand"))
				var/mutable_appearance/bloodsies_right = mutable_appearance(dna.species.blood_mask, "bloodyhand_right", layer = -R_HAND_BLOOD_LAYER)
				bloodsies_right.color = feet_blood_color
				overlays_standing[R_HAND_BLOOD_LAYER] = bloodsies_right
```

File: code/modules/mob/living/carbon/human/human_update_icons.dm [hunk 4]
```diff
@@ -788,19 +799,26 @@ GLOBAL_LIST_EMPTY(damage_icon_parts)
...
-			overlays_standing[SHOES_LAYER] = bloodsies
+		if(feet_blood_DNA) // Checks for feet to make sure we don't have mysterious floating blood.
+			if(get_organ("l_foot"))
+				var/mutable_appearance/bloodsies_left 	= mutable_appearance(dna.species.blood_mask, "shoeblood_left", layer = -L_FOOT_BLOOD_LAYER)
+				bloodsies_left.color = feet_blood_color
+				overlays_standing[L_FOOT_BLOOD_LAYER] 	= bloodsies_left
+			if(get_organ("r_foot"))
+				var/mutable_appearance/bloodsies_right 	= mutable_appearance(dna.species.blood_mask, "shoeblood_right", layer = -R_FOOT_BLOOD_LAYER)
+				bloodsies_right.color = feet_blood_color
+				overlays_standing[R_FOOT_BLOOD_LAYER] 	= bloodsies_right
```
Comment (lewcc) [inline code/modules/mob/living/carbon/human/human_update_icons.dm:818]:
```suggestion
		if(feet_blood_DNA) // Checks for feet to make sure we don't have mysterious floating blood.
			if(get_organ("l_foot"))
				var/mutable_appearance/bloodsies_left = mutable_appearance(dna.species.blood_mask, "shoeblood_left", layer = -L_FOOT_BLOOD_LAYER)
				bloodsies_left.color = feet_blood_color
				overlays_standing[L_FOOT_BLOOD_LAYER] = bloodsies_left
			if(get_organ("r_foot"))
				var/mutable_appearance/bloodsies_right = mutable_appearance(dna.species.blood_mask, "shoeblood_right", layer = -R_FOOT_BLOOD_LAYER)
				bloodsies_right.color = feet_blood_color
				overlays_standing[R_FOOT_BLOOD_LAYER] = bloodsies_right
```

File: code/__DEFINES/misc_defines.dm (hunk 2 above)
Comment (Spaghetti-bit) [reply code/__DEFINES/misc_defines.dm]: Fixed with 
image

**Review by lewcc — APPROVED** [review]
LGTM

**Pull request merged**